presidio-analyzer/tests/test_analyzer_engine.py

```diff
@@ -265,3 +285,73 @@ def test_when_allFields_is_true_and_entities_not_empty_exception(self):
         new_field.minScore = '0.5'
         with pytest.raises(ValueError):
             analyze_engine.Apply(request, None)
+
+    # Context tests
+    def test_text_with_context_improves_score(self):
+        test_items = self.load_test_items()
+
+        # Currently we have 19 sentences, this is a sanity
+        assert(len(test_items) == 19)
+        analyze_engine = AnalyzerEngine()
+        nlp_engine = SpacyNlpEngine()
+        
+        for item in test_items:
+            text = item[0]
+            recognizer = item[1]
+            entities = item[2]
+            nlp_artifacts = nlp_engine.process_text(text, "en")
+            results = recognizer.analyze(text, entities)
+
+            enhanced_results = analyze_engine.enhance_using_context(text,
+                                                                    recognizer,
+                                                                    results,
+                                                                    nlp_artifacts)
+            assert(results)
+            assert(len(results) == len(enhanced_results))
+            for i in range(len(results)):
+                assert(results[i].score < enhanced_results[i].score)
+
+    def load_test_items(self):
```

Comment:
This is a test fixture - it prepares data for a test.
Since this is a static data, it should be instantiated once before the tests begin.
Since this data may be share with other UT, you should consider using the setUp() and tearDown() that will be executed before and after each test method.

Reply:
👍